server/src/catalog.rs

```diff
@@ -296,47 +305,113 @@ pub struct NewModelTableMetadata {
+                // If the fields of the tag columns is not of type Utf8, return an error.
+                if !tag_field.data_type().equals_datatype(&DataType::Utf8) {
+                    return Err(ModelarDBError::ConfigurationError(format!(
+                        "The tag column with index '{}' is not of type '{}'.",
+                        tag_column_index,
+                        DataType::Utf8
+                    )));
+                }
+            } else {
+                // If the indices for the tag columns does not match the schema, return an error.
+                return Err(ModelarDBError::ConfigurationError(format!(
+                    "The tag column index '{}' does not match a field in the schema.",
+                    tag_column_index
+                )));
+            }
+        }
+
+        let field_column_indices: Vec<usize> = (0..schema.fields().len()).filter(|index| {
+            // TODO: Change this cast when indices in the action body are changed to use two bytes.
+            let index = *index as u8;
+            index != timestamp_column_index && !tag_column_indices.contains(&index)
+        }).collect();
+
+        // If there are no field columns, return an error.
+        if field_column_indices.is_empty() {
+            return Err(ModelarDBError::ConfigurationError(
+                "There needs to be at least one field column.".to_owned()
+            ));
+        } else {
+            for field_column_index in &field_column_indices {
+                // unwrap() is safe to use since the indices are collected from the schema fields.
+                let field = schema.fields.get(*field_column_index).unwrap();
+
+                // If the fields of the field columns is not of type ArrowValue, return an error.
+                if !field.data_type().equals_datatype(&ArrowValue::DATA_TYPE) {
+                    return Err(ModelarDBError::ConfigurationError(format!(
+                        "The field column with index '{}' is not of type '{}'.",
+                        field_column_index,
+                        ArrowValue::DATA_TYPE
+                    )));
+                }
+            }
+        }
 
         // If there are duplicate tag columns, return an error. HashSet.insert() can be used to check
         // for uniqueness since it returns true or false depending on if the inserted element already exists.
         let mut uniq = HashSet::new();
-        if !tag_column_indices.clone().into_iter().all(|x| uniq.insert(x)) {
-            return Err("The tag column indices cannot have duplicates.".to_owned());
+        if !tag_column_indices
+            .clone()
+            .into_iter()
+            .all(|x| uniq.insert(x))
+        {
+            return Err(ModelarDBError::ConfigurationError(
+                "The tag column indices cannot have duplicates.".to_owned()
+            ));
         }
 
         // If there are more than 1024 columns, return an error. This limitation is necessary
```

Comment:
As we discussed it might be problematic to change the key structure in the future since it leads to difficulties with referencing data that was created using the old key structure. It might therefore also be unnecessary to make changes to the code that makes it easier to change the key structure.

Reply:
I agree that it would be very complicated to change from 64 bit to something else for existing data. I was more thinking if someone ever would want to have different limits from the beginning and compile with other values. Maybe not very realistic...